### ISSUE TITLE
Add the possibility to set a "priority" value to issues

### ISSUE THREAD

**Issue by Delgan** [isaacs/github#472]
Currently, we can apply several tags to an issue, we can set it to a milestone, we can assign someone to it, but there is no way to set a priority.

Many issues trackers on others website have this feature, it is very useful and I am surprised that Github issues do not have such option.

Of course, we can use tags.
But this makes the issues list much harder to browse, it decreases readability as it is complicated to differentiate regular tag and priority tag.
Moreover, tags are used to see what kind of issue it is and to give specific additional information about its content. Priority does not really give inherent information about the issue, it is mainly used to manage the problem relatively to the others issues.

I really think that a light priority system should be added to Github issues tracker.

**Comment by cirosantilli** [comment]
Possible duplicate of: https://github.com/isaacs/github/issues/260

**Comment by bgvo** [comment]
+1

**Comment by ReanimationXP** [comment]
+1

**Comment by HarveyEV** [comment]
+1

**Comment by ReanimationXP** [comment]
Would also like to note that one cannot sort on tags.

**Comment by cruoti** [comment]
+1

**Comment by jamesui** [comment]
+1

**Comment by sangallidavide** [comment]
+1

**Comment by AlecBenzer** [comment]
@cirosantilli I'm not sure I'd call this a dup of #260. That seems to be about inducing a priority ordering on issues by being able to re-order them as a list. This seems to be more about explicitly assigning priorities to issues, probably from a set list of something like P1, P2, P3, P4, P5.

**Comment by abrambailey** [comment]
+1

**Comment by AlecBenzer** [comment]
Please don't reply to an issue just to say '+1': it sends a notification to everyone subscribed to the issue. Instead, just react to the issue's initial comment with the 👍 emoji.

<img width="233" alt="plus_1_github" src="https://user-images.githubusercontent.com/92369/31297680-0455e07c-aab5-11e7-81dc-d6070ca75378.png">

**Comment by andzandz** [comment]
Agreed, would be very useful for the personal issues page https://github.com/issues to be able to sort by priority; i.e. show high priority issues across all projects together.

**Comment by FlorianBruniaux** [comment]
+1

**Comment by andzandz** [comment]
I've found a way to do this, if you put a label called "priority" on all priority issues, you can search for this at https://www.github.com/issues (for organisation or user "userxyz")

```
is:open is:issue label:priority sort:project user:userxyz
```

I'd like a divider between the priority issues of each project listed, but it's a start

**Comment by ReanimationXP** [comment]
Labels isn't a solution, please stop presenting it as such.

**Comment by artieshevchenko** [comment]
+1

**Comment by uneeb123** [comment]
+1

**Comment by aowss** [comment]
+1

**Comment by mchavarriae** [comment]
+1

**Comment by andzandz** [comment]
@artieshevchenko @uneeb123 @aowss @mchavarriae 
https://github.com/isaacs/github/issues/472#issuecomment-334864705

**Comment by abrambailey** [comment]
@artieshevchenko The reason people +1 is to bump visibility.

**Comment by AlecBenzer** [comment]
@abrambailey in what way does commenting "+1" bump visibility?

**Comment by abrambailey** [comment]
> Also, it lets latecomers (like me) know that people are actively interested in this feature. If I saw that there was a post from 2015 with 60 likes, I would probably consider the issue ignored/dead unless I saw recent comment activity.

Sounds like another good feature request. How about a dedicated issue up / downvote with aggregate vote timeline / line graph?

**Comment by milescrawford** [comment]
@AlecBenzer By making the issue appear in everyone's inboxes. Squeaky wheel gets the grease.

**Comment by MgSam** [comment]
I imagine many Github projects use Issues in lieu of an external bug tracking system. So why doesn't Github make Issues competitive with a full featured bug tracker? For us, it is a major differentiator between Atlassian (Bitbucket and Jira) vs Github.

The Issues board quickly becomes noise without having priorities.

**Comment by C-Kenny** [comment]
+1

**Comment by mp-vayyar** [comment]
+1

**Comment by ghu3** [comment]
+1

**Comment by cmpickle** [comment]
+1

**Comment by pklepacki** [comment]
Any new info on this after 2 years?

**Comment by Neuroforge** [comment]
Why don't you use your own labels? 

Labels -> Edit.

**Comment by pklepacki** [comment]
> Why don't you use your own labels?

@Neuroforge I believe these are specific enough that separate UI element would be required. F.ex. opened/closed issues also could be accomplished by labels, yet there are green/red exclamation marks on issues' cards. Also setting this as custom labels is not that intuitive - this thread is here for a reason. Not to mention reasons stated in previous comments...

**Comment by HarveyEV** [comment]
Agreed. As for us, we already rely on labels for too much. Its still a good idea two years later despite me following up and pinging on it every few months.

**Comment by suenchunhui** [comment]
This is possible using zenhub and github enterprise, which is a paid thing. direct competition to their biz model?

**Comment by andrewschultz** [comment]
A workaround is to have labels saying "1 Critical," "2 Blocker", "3 Major", "4 Minor", "5 Trivial" to ape Bitbucket. Then when the default sort (for the first label alphabetically) will show the critical issues first.

PyGithub can run a relatively simple script to track this. 

```from github import Github

gh = Github(user_name, user_password)
user = gh.get_user()
repo = user.get_repo(my_repo)
prio_list = [ x.name for x in repo.get_labels() if re.search("^[1-5] ", x.name) ]
open_issues = repo.get_issues()
for issue in open_issues:
    label_list = [x.name for x in issue.labels] # The rest depends on how you want the output format to work. Match prio_list to label_list.
```

I suspect that being able to do this with full speed and convenience is what more complex repo owners pay GitHub for, and the saved time is worth it to them.

Individual users get enough for free! And PyGithub is well worth learning. So my workaround is good enough for me.

**Comment by kajmagnus** [comment]
I'm curious as to why peopleo don't use Milestones? As far as I've read, if you create Milestones, and add issues to a Milestone, then you can sort them by priority, inside the Milestone view.

Look, a video: https://github.blog/2016-07-05-reorder-issues-within-a-milestone/, and:

> You can now reorder issues and pull requests and indicate priority by moving them higher up or lower down the list

<small>(Disclosure: I don't use GitHub issues at all, for other reasons, for my projects.)</small>

**Comment by patspam** [comment]
I was going to say it's easy, just embed priority in the Title field, eg. `[P0] Build website`

..except that GitHub issues doesn't support sorting by Title... 🤦‍♂

**Comment by daobrien** [comment]
> I'm curious as to why peopleo don't use Milestones and their related priority features? As far as I've read, if you create Milestones, and add issues to a Milestone, then you can sort them by priority, inside the Milestone view.
> 
> Look, a video: https://github.blog/2016-07-05-reorder-issues-within-a-milestone/, and:
> 
> > You can now reorder issues and pull requests and indicate priority by moving them higher up or lower down the list  [within the milestone]
> 
> (Disclosure: I don't use GitHub issues at all, for other reasons, for my projects.)

afaict this no longer works. At least, it doesn't work for me. The little indicator bars on the left side of each issue do not appear. There seems to be no way to move them. Kbd shortcuts don't work either.
I would very much like to be able to apply and sort by priority.

**Comment by kamyker** [comment]
Use Project Boards :)

Set ToDo list to receive all new/reopened issues
Set Done list to receive all closed issues

Use "Add Card" to quickly drag and drop all cards that are not added to the project

**Comment by cheater** [comment]
+1 just do it. Maybe add the ability to have a tag that has multiple values, and only one of them can be present on any issue. Maybe do it some other way. But this has been requested **in the first half of the last decade** in #260.

**Comment by dereks** [comment]
> But this has been requested in the first half of the last decade

Priorities used to be an Issues feature. GitHub removed them in early 2011. So they should probably close this issue as Won't Fix, right?

https://blossom.co/index.html%3Fp=148.html

https://github.blog/2011-04-09-issues-2-0-the-next-generation/

**Comment by cheater** [comment]
> Priorities used to be an Issues feature. GitHub removed them in early 2011. So they should probably close this issue as Won't Fix, right?


This feature request isn't about what github did or is doing, it's about what users want github to do.

**Comment by alter** [comment]
I'd like to get it too...

**Comment by Displee** [comment]
+1

**Comment by Kevin-CodeCrew** [comment]
+1
